src/saml2/config.py

```diff
@@ -569,13 +585,198 @@ def ecp_endpoint(self, ipaddress):
+
+    def get_application_identifier(self):
+        return getattr(self, "_sp_application_identifier", None)
+
+    def get_protocol_version(self):
+        return getattr(self, "_sp_protocol_version", None)
+
+    def get_node_country(self):
+        return getattr(self, "_sp_node_country", None)
+
+    @property
+    def warning_validators(self):
+        sp_warning_validators = {
+            "hide_assertion_consumer_service SHOULD be set to True":
+                getattr(self, "_sp_hide_assertion_consumer_service", None) is True
+        }
+        return {**super().warning_validators, **sp_warning_validators}
+
+    @property
+    def error_validators(self):
+        sp_error_validators = {
+            "authn_requests_signed MUST be set to True":
+                getattr(self, "_sp_authn_requests_signed", None) is True,
+            "sp_type MUST be set to 'public' or 'private'":
+                getattr(self, "_sp_sp_type", None) in ("public", "private")
+        }
+        return {**super().error_validators, **sp_error_validators}
+
+
 class IdPConfig(Config):
     def_context = "idp"
 
     def __init__(self):
         Config.__init__(self)
 
 
+class eIDASIdPConfig(IdPConfig, eIDASConfig):
+    def get_endpoint_element(self, element):
+        return getattr(self, "_idp_endpoints", {}).get(element, None)
+
+    def get_application_identifier(self):
+        return getattr(self, "_idp_application_identifier", None)
+
+    def get_protocol_version(self):
+        return getattr(self, "_idp_protocol_version", None)
+
+    def get_node_country(self):
+        return getattr(self, "_idp_node_country", None)
+
+    def verify_non_notified_loa(self):
+        return not any(
+            [x.startswith("http://eidas.europa.eu/LoA/")
+             for x in getattr(self, "_idp_supported_loa", {}).get("non_notified")]
+        )
+
+    def verify_notified_loa(self):
+        return all(
+            [x in ["http://eidas.europa.eu/LoA/low",
+                   "http://eidas.europa.eu/LoA/substantial",
+                   "http://eidas.europa.eu/LoA/high"]
+             for x in getattr(self, "_idp_supported_loa", {}).get("notified")]
+        )
+
+    @property
+    def warning_validators(self):
+        idp_warning_validators = {}
+        return {**super().warning_validators, **idp_warning_validators}
+
+    @property
+    def error_validators(self):
+        idp_error_validators = {
+            "want_authn_requests_signed MUST be set to True":
+            getattr(self, "_idp_want_authn_requests_signed", None) is True,
+            "provided_attributes MUST be set to denote the supported attributes by "
+            "the IdP":
+            not_empty(getattr(self, "_idp_provided_attributes", None)),
+            "supported_loa for non-notified eIDs MUST NOT use an "
+            "http://eidas.europa.eu/LoA/ prefix":
+            self.verify_non_notified_loa(),
+            "supported_loa for notified eID MUST be (at least) one of "
```

Comment:
Doesn't eIDAS allow for non-notified eIDs? Why do we error in this (and previous) case?

Reply:
It allows optional support for non-notified eIDs as long as they don't start with the `http://eidas.europa.eu/LoA/` prefix and MUST support the eIDAS Levels of Assurance (LoA) as defined in Section 3.2

So that's what we're doing in these two handlers
1. We make sure that the LoAs for non-notified eIDs don't use the "blacklisted" prefix
2. We make sure that the LoAs for the notified eIDs are some of those that are defined in the spec document section 3.2 (low/substantial/high)